ledger/participant-integration-api/src/main/scala/platform/store/backend/common/CompletionStorageBackendTemplate.scala

```diff
@@ -60,26 +56,41 @@ class CompletionStorageBackendTemplate(
           participant_command_completions
         WHERE
           ${queryStrategy.offsetIsBetween(
-          nonNullableColumn = "completion_offset",
-          startExclusive = startExclusive,
-          endInclusive = endInclusive,
-        )} AND
-          application_id = $applicationId AND
-          ${queryStrategy.arrayIntersectionNonEmptyClause("submitters", internedParties)}
-        ORDER BY completion_offset ASC"""
-        .as(completionParser.*)(connection)
+        nonNullableColumn = "completion_offset",
+        startExclusive = startExclusive,
+        endInclusive = endInclusive,
+      )} AND
+          application_id = $applicationId
+        ORDER BY completion_offset ASC
+        ${QueryStrategy.limitClause(Some(limit))}"""
+      .asVectorOf(completionParser)(connection)
+    val filteredBySubmitters = result.collect {
+      case (submitters, response) if submitters.exists(internedParties) => response
```

Comment:
please keep scope of StorageBackend clear: JVM filtration by interned parties is a concern for the CommandCompletionsReader

Reply:
Discussed offline. For the record: keeping filtration in the backend template